src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.ValueTaskSource.cs

```diff
@@ -29,42 +29,37 @@ private unsafe class ValueTaskSource : IValueTaskSource<int>, IValueTaskSource
             private bool _cancellationHasBeenRegistered;
 #endif
 
-            public static ValueTaskSource Create(
-                AsyncWindowsFileStreamStrategy strategy,
-                PreAllocatedOverlapped? preallocatedOverlapped,
-                ReadOnlyMemory<byte> memory)
-            {
-                // If the memory passed in is the strategy's internal buffer, we can use the base AwaitableProvider,
-                // which has a PreAllocatedOverlapped with the memory already pinned.  Otherwise, we use the derived
-                // MemoryAwaitableProvider, which Retains the memory, which will result in less pinning in the case
-                // where the underlying memory is backed by pre-pinned buffers.
-                return preallocatedOverlapped != null &&
-                       MemoryMarshal.TryGetArray(memory, out ArraySegment<byte> buffer) &&
-                       preallocatedOverlapped.IsUserObject(buffer.Array) ?
-                            new ValueTaskSource(strategy, preallocatedOverlapped, buffer.Array) :
-                            new MemoryValueTaskSource(strategy, memory);
-            }
+            internal ValueTaskSource(AsyncWindowsFileStreamStrategy strategy)
+                => _strategy = strategy;
 
-            protected ValueTaskSource(
-                AsyncWindowsFileStreamStrategy strategy,
-                PreAllocatedOverlapped? preallocatedOverlapped,
-                byte[]? bytes)
+            internal ValueTaskSource(AsyncWindowsFileStreamStrategy strategy, byte[] internalBuffer) : this(strategy)
+                => _preallocatedOverlapped = new PreAllocatedOverlapped(s_ioCallback, this, internalBuffer);
```

Comment:
Can we create the preallocated overlapped now even if there is no internal buffer?  That should make the no buffering case measurably faster, yes?

Reply:
> Can we create the preallocated overlapped now even if there is no internal buffer? That should make the no buffering case measurably faster, yes?

This is a very good quesiton. Let me give it a try